### PR TITLE
CRM-20969 Fix issues in order by clauses of reports showing errors in 7.1

### PR DESCRIPTION
**Pull request by seamuslee001**
* [CRM-20969: Fix issue in reports where we try and append a new array key to string](https://issues.civicrm.org/jira/browse/CRM-20969)

### CODE REVIEW

**Comment by seamuslee001** [comment]
ping @monishdeb @eileenmcnaughton does this make sense to both of you?

**Comment by eileenmcnaughton** [comment]
@seamuslee001 - alternative proposal https://github.com/civicrm/civicrm-core/pull/10779

File: CRM/Report/Form/Walklist/Walklist.php [hunk 1]
```diff
@@ -227,11 +227,11 @@ public function orderBy() {
     foreach ($this->_columns as $tableName => $table) {
       if (array_key_exists('order_bys', $table)) {
         foreach ($table['order_bys'] as $fieldName => $field) {
-          $this->_orderBy[] = $field['dbAlias'];
```
Comment (eileenmcnaughton) [inline CRM/Report/Form/Walklist/Walklist.php]: It should be possible to simply remove this function

File: CRM/Report/Form/Pledge/Summary.php [hunk 2]
```diff
@@ -268,25 +268,25 @@ public function groupBy() {
                 )) {
                   $append = '';
                 }
-                $this->_groupBy[] = "$append {$this->_params['group_bys_freq'][$fieldName]}({$field['dbAlias']})";
```
Comment (eileenmcnaughton) [inline CRM/Report/Form/Pledge/Summary.php]: hopefully this whole function can go too

File: CRM/Report/Form/Walklist/Walklist.php (hunk 1 above)
Comment (seamuslee001) [inline CRM/Report/Form/Walklist/Walklist.php]: I don't want to try that in this PR as i think that is a separate issue to deal with, This is just about making sure these reports pass php7.1

File: CRM/Report/Form/Pledge/Summary.php (hunk 2 above)
Comment (seamuslee001) [inline CRM/Report/Form/Pledge/Summary.php]: Same as above

Comment (eileenmcnaughton) [inline CRM/Report/Form/Pledge/Summary.php]: Hmm - If I'm going to test out these reports I'd rather test the 'right' change.  I just tested the Walklist report - with the  existing (non 7.1 compliant) order by function sort by ascending works, sort by descending does not work, section break does not work. Removing the order by function causes it to work correctly (and fixes your php problem) - note the walklist report needs to be enabled under manage templates.

**Comment by seamuslee001** [comment]
@eileenmcnaughton i have removed the orderBy Function given your testing, The groupBy ones seem to be more indivualised to the reports so i think they should be held onto, Only the walklist had issues with orderBy the others had issues with groupBy

**Comment by eileenmcnaughton** [comment]
Actually they are not so much individualised as copy & pasted at different stages of development :-). If we overwrote the parent groupBy with the one from pledgeSummary we could remove it from most reports. However, I think we have met the requirement of including some code improvement in this PR so I'm OK to merge now

**Comment by monishdeb** [comment]
Merging as per tag

**Pull request merged**